File: WEEK7/Python/bj13_2166.py
import sys

n = int(sys.stdin.readline())
x = []
y = []

for _ in range(n):
    a, b = map(int, sys.stdin.readline().split())
    x.append(a)
    y.append(b)

x.append(x[0])
y.append(y[0])

sum_a = 0
sum_b = 0

for i in range(n):
    sum_a += x[i] * y[i + 1]
    sum_b += y[i] * x[i + 1]

print(round(abs(sum_a - sum_b) / 2, 1))

File: WEEK7/Java/bj13_2166.java
import java.io.*;
import java.util.*;

public class bj13_2166 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        long[][] p = new long[n + 1][2];

        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            p[i][0] = Long.parseLong(st.nextToken());
            p[i][1] = Long.parseLong(st.nextToken());
        }
        p[n][0] = p[0][0];
        p[n][1] = p[0][1];

        long sumA = 0;
        long sumB = 0;

        for (int i = 0; i < n; i++) {
            sumA += p[i][0] * p[i + 1][1];
            sumB += p[i][1] * p[i + 1][0];
        }

        System.out.printf("%.1f", Math.abs(sumA - sumB) / 2.0);
    }
}